test/integration/roles/test_ecs_ecr/tasks/main.yml

```diff
@@ -5,6 +5,10 @@
 - block:
     - name: When creating with check mode
       ecs_ecr: name='{{ ecr_name }}' region='{{ ec2_region }}'
+      environment:
```

Comment:
@michael-dev2rights @willthames Retracting my thoughts on this. Until we've documented the process for switching to using profile we don't really have another choice. But we shouldn't use environment variables unless we're testing them. Passing the credentials to the module is fine. I'll work on fixing those.